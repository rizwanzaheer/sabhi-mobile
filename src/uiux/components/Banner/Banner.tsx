import * as React from 'react'
import { ImageBackground, ViewStyle } from 'react-native'
import Container from '../Container/Container'
import Text, { TextTypes } from '../Text/Text'
import Avatar from '../Avatar/Avatar'
import * as Kancha from '../../types'
import { withTheme } from '../../theming'

interface BannerProps {
  /**
   * Background image for banner
   */
  backgroundImage?: any

  /**
   * Banner size
   */
  size?: 'small' | 'medium'

  /**
   * Background color
   */
  backgroundColor?: string

  /**
   * Issuer
   */
  issuer: Kancha.Identity

  /**
   * Title of banner
   */
  title: string

  /**
   * Subtitle to show under title
   */
  subTitle?: string

  /**
   * Theme
   */
  theme: any
}

const bannerSize: { [index: string]: number } = {
  small: 180,
  medium: 250,
}

const Banner: React.FC<BannerProps> = props => {
  const BannerStyle: ViewStyle = {
    backgroundColor: '#333333',
    flex: 1,
    height: props.size ? bannerSize[props.size] : bannerSize.medium,
  }
  const source = props.issuer.profileImage ? { source: { uri: props.issuer.profileImage } } : {}
  return (
    <ImageBackground source={props.backgroundImage} style={BannerStyle}>
      <Container
        backgroundColor={props.backgroundColor ? props.backgroundColor : 'rgba(0,0,0,0.8)'}
        flex={1}
        alignItems={'center'}
        justifyContent={'center'}
        paddingTop={true}
      >
        <Avatar {...source} size={56} type={'rounded'} gravatarType={'retro'} address={props.issuer.did} />
        <Container paddingTop={true} alignItems={'center'}>
          <Text type={TextTypes.H2} bold={true} textColor={'#FFFFFF'}>
            {props.title}
          </Text>
          <Container paddingTop={5} alignItems={'center'}>
            <Text type={TextTypes.SubTitle} textColor={'#FFFFFF'}>
              {props.subTitle && props.subTitle}
            </Text>
          </Container>
        </Container>
      </Container>
    </ImageBackground>
  )
}

export default withTheme(Banner)
