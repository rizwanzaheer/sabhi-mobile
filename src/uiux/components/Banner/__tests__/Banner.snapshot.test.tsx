import * as React from 'react'
import { render } from 'react-native-testing-library'
import Banner from '../Banner'

describe('Component(snapshot): Banner', () => {
  it('should render with default props', () => {
    const tree = render(
      <Banner
        title={'Test Title'}
        issuer={{ did: 'ethr:did', shortId: 'Test name' }}
        subTitle={'Test Subtitle'}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with default props', () => {
    const tree = render(
      <Banner
        size={'medium'}
        issuer={{ did: 'ethr:did', shortId: 'Test name' }}
        backgroundImage={{ uri: '' }}
        backgroundColor={'#000000'}
        title={'Test Title'}
        subTitle={'Test Subtitle'}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
