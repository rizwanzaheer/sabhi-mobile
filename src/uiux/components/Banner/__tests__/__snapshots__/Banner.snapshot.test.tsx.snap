// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Banner should render with default props 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#333333",
      "flex": 1,
      "height": 250,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 250,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.8)",
        "borderRadius": undefined,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": 15,
        "width": undefined,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.gravatar.com/avatar/08e65c6403d55f307a163a5d7e4a97cf?s=112&d=retro",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#0078ff",
            "borderRadius": 5,
            "height": 56,
            "width": 56,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": 15,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flexWrap": "wrap",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Test Title
      </Text>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": 5,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "flexWrap": "wrap",
              "fontSize": 14,
            }
          }
        >
          Test Subtitle
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component(snapshot): Banner should render with default props 2`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#333333",
      "flex": 1,
      "height": 250,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 250,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": undefined,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": 15,
        "width": undefined,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.gravatar.com/avatar/08e65c6403d55f307a163a5d7e4a97cf?s=112&d=retro",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#0078ff",
            "borderRadius": 5,
            "height": 56,
            "width": 56,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": 15,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "flexWrap": "wrap",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Test Title
      </Text>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": 5,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "flexWrap": "wrap",
              "fontSize": 14,
            }
          }
        >
          Test Subtitle
        </Text>
      </View>
    </View>
  </View>
</View>
`;
